src/sentry/snuba/tasks.py

```diff
@@ -195,3 +201,36 @@ def _delete_from_snuba(dataset, subscription_id):
     )
     if response.status != 202:
         raise SnubaError("HTTP %s response from Snuba!" % response.status)
+
+
+@instrumented_task(
+    name="sentry.snuba.tasks.subscription_checker", queue="subscriptions",
+)
+def subscription_checker(**kwargs):
```

Comment:
Why do subscriptions get stuck and need this cleanup?

Reply:
We fire off a separate task to create the subscription in snuba. I'm not totally sure why they fail, but I suspect it's likely due to a failure on the call to snuba, maybe a network burp or something along those lines. This happens pretty rarely in general.

Reply:
ok. I ask because repair tasks can sometimes paper over actual bugs. If we're working around distributed failures, then repairing afterwards makes sense.

Reply:
Yeah, I suspect it's entirely distributed failures here. I've added stats in so that we can see how many times we're doing this, ideally it's infrequent